stl/inc/span

```diff
@@ -0,0 +1,778 @@
+            static_assert(_Extent > 0, "back of empty span");
+        }
+#if _CONTAINER_DEBUG_LEVEL > 0
+        else {
+            _STL_VERIFY(size() > 0, "back of empty span");
+        }
+#endif // _CONTAINER_DEBUG_LEVEL > 0
+        return _Mydata[size() - 1];
+    }
+
+    _NODISCARD constexpr pointer data() const noexcept {
+        return _Mydata;
+    }
+
+    // [span.iterators] Iterator support
+    _NODISCARD constexpr iterator begin() noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        return {_Mydata, _Mydata, _Mydata + size()};
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+        return {_Mydata};
+#endif // _ITERATOR_DEBUG_LEVEL
+    }
+
+    _NODISCARD constexpr iterator end() noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        return {_Mydata + size(), _Mydata, _Mydata + size()};
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+        return {_Mydata + size()};
+#endif // _ITERATOR_DEBUG_LEVEL
+    }
+
+    _NODISCARD constexpr const_iterator begin() const noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        return {_Mydata, _Mydata, _Mydata + size()};
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+        return {_Mydata};
+#endif // _ITERATOR_DEBUG_LEVEL
+    }
+
+    _NODISCARD constexpr const_iterator end() const noexcept {
+#if _ITERATOR_DEBUG_LEVEL >= 1
+        return {_Mydata + size(), _Mydata, _Mydata + size()};
+#else // ^^^ _ITERATOR_DEBUG_LEVEL >= 1 ^^^ // vvv _ITERATOR_DEBUG_LEVEL == 0 vvv
+        return {_Mydata + size()};
+#endif // _ITERATOR_DEBUG_LEVEL
+    }
+
+    _NODISCARD constexpr const_iterator cbegin() const noexcept {
+        return begin();
+    }
+
+    _NODISCARD constexpr const_iterator cend() const noexcept {
+        return end();
+    }
+
+    _NODISCARD constexpr reverse_iterator rbegin() noexcept {
+        return reverse_iterator{end()};
+    }
+
+    _NODISCARD constexpr reverse_iterator rend() noexcept {
+        return reverse_iterator{begin()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator rbegin() const noexcept {
+        return const_reverse_iterator{end()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator rend() const noexcept {
+        return const_reverse_iterator{begin()};
+    }
+
+    _NODISCARD constexpr const_reverse_iterator crbegin() const noexcept {
+        return rbegin();
+    }
+
+    _NODISCARD constexpr const_reverse_iterator crend() const noexcept {
+        return rend();
+    }
+
+    _NODISCARD constexpr pointer _Unchecked_begin() noexcept {
+        return _Mydata;
+    }
+
+    _NODISCARD constexpr const_pointer _Unchecked_begin() const noexcept {
+        return _Mydata;
+    }
+
+    _NODISCARD constexpr pointer _Unchecked_end() noexcept {
+        return _Mydata + size();
+    }
+
+    _NODISCARD constexpr const_pointer _Unchecked_end() const noexcept {
+        return _Mydata + size();
+    }

```

Comment:
Although these functions are internal, they should behave like begin/end with respect to shallow constness. So, they should have only `const noexcept` overloads returning `pointer`.

Reply:
Done